src/jobservice/logger/sweeper/db_sweeper.go

```diff
@@ -0,0 +1,57 @@
+package sweeper
+
+import (
+	"time"
+	"fmt"
+	"github.com/goharbor/harbor/src/common/dao"
+)
+
+var dbInit = make(chan int, 1)
+var isDBInit = false
+
+// DBSweeper is used to sweep the DB logs
+type DBSweeper struct {
+	duration int
+}
+
+// NewDBSweeper is constructor of DBSweeper
+func NewDBSweeper(duration int) *DBSweeper {
+	return &DBSweeper{
+		duration: duration,
+	}
+}
+
+// Sweep logs
+func (dbs *DBSweeper) Sweep() (int, error) {
+	// DB initialization not completed, waiting
+	WaitingDBInit()
+
+	// Start to sweep logs
+	before := time.Now().Add(time.Duration(dbs.duration) * oneDay * -1)
```

Comment:
time.Now().Sub?

Reply:
sub accept  a parameter Time and return Duration, not applicable.

Reply:
correct, my bad